OData/test/E2ETest/WebStack.QA.Test.OData/ODataCountTest/CountTest.cs

```diff
@@ -0,0 +1,105 @@
+
+            configuration.Routes.Clear();
+            HttpServer httpServer = configuration.GetHttpServer();
+            configuration.MapODataServiceRoute(
+                routeName: "odata",
+                routePrefix: "odata",
+                model: CountEdmModel.GetModel());
+
+            configuration.EnsureInitialized();
+        }
+
+        [Theory]
+        [InlineData("{0}/odata/Heroes/$count", 1)]
+        [InlineData("{0}/odata/Heroes/Default.GetWeapons()/$count", 5)]
+        [InlineData("{0}/odata/Heroes/Default.GetNames()/$count", 2)]
+        public async Task DollarCountWorksWithEF(string url, int expectedCount)
+        {   // Arrange
+            string requestUri = string.Format(url, BaseAddress);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            int actualCount = Int32.Parse(response.Content.ReadAsStringAsync().Result);
+            Assert.Equal(expectedCount, actualCount);
+        }
+
+        [Theory]
+        [InlineData("{0}/odata/Heroes?$count=true", 1)]
+        [InlineData("{0}/odata/Heroes/Default.GetWeapons()?$count=true", 5)]
+        [InlineData("{0}/odata/Heroes/Default.GetNames()?$count=true", 2)]
+        public async Task CountQueryOptionWorksWithEF(string url, int expectedCount)
+        {   // Arrange
+            string requestUri = string.Format(url, BaseAddress);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            JObject result = response.Content.ReadAsAsync<JObject>().Result;
+            Assert.Equal(expectedCount, result["@odata.count"]);
+        }
+
+        [Theory]
+        [InlineData("{0}/odata/Heroes")]
+        [InlineData("{0}/odata/Heroes/Default.GetWeapons()")]
+        [InlineData("{0}/odata/Heroes/Default.GetNames()")]
+        [InlineData("{0}/odata/Heroes?$count=false")]
+        [InlineData("{0}/odata/Heroes/Default.GetWeapons()?$count=false")]
+        [InlineData("{0}/odata/Heroes/Default.GetNames()?$count=false")]
+        public async Task NegativeCountTest(string url)
+        {   // Arrange
+            string requestUri = string.Format(url, BaseAddress);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            string result = response.Content.ReadAsStringAsync().Result;
+            int tryParseInt;
+            bool tryResult = Int32.TryParse(result, out tryParseInt);
```

Comment:
why do you try to parse the total payload? As I know, you should only to test the @odata.count.